namerd/core/src/main/scala/io/buoyant/namerd/DtabStore.scala

```diff
@@ -18,6 +18,12 @@ trait DtabStore {
   def create(ns: String, dtab: Dtab): Future[Unit]
 
   /**
+   * Deletes a dtab.  Returns a DtabNamespaceDoesNotExistException if the
+   * namespace does not exist.
+   */
+  def delete(ns: String): Future[Unit]
```

Comment:
There's some discussion of that here: https://github.com/BuoyantIO/linkerd/issues/207.  Ideally yes, though the cost to implementing this depends on the backing store.